python/pyarrow/tests/test_dataset.py

```diff
@@ -4594,6 +4604,7 @@ def test_write_table_multiple_fragments(tempdir):
     )
 
 
+@pytest.mark.numpy
 def test_write_iterable(tempdir):
     table = pa.table([
         pa.array(range(20)), pa.array(np.random.randn(20)),
```

Comment:
Some of those `np.random.randn(20)` calls could be replaced with an utility using the stdlib? (can also be for a follow-up)